x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/MultiRowTestCaseSupplier.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.test.ESTestCase.randomBoolean;
+import static org.elasticsearch.test.ESTestCase.randomList;
+import static org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier.TypedDataSupplier;
+
+/**
+ * Extension of {@link TestCaseSupplier} that provided multi-row test cases.
+ */
+public final class MultiRowTestCaseSupplier {
+
+    private MultiRowTestCaseSupplier() {}
+
+    public static List<TypedDataSupplier> multiRowIntCases(int minRows, int maxRows, int min, int max, boolean includeZero) {
```

Comment:
I'd probably just call this `intCases` because it's in the `MutliRowTestCaseSupplier` class. Even if we do a `static` import I think it'll be clear enough to readers. Maybe not super casual readers, but I think clear enough.